axelrod/tests/unit/test_titfortat.py

```diff
@@ -497,4 +497,25 @@ def test_world_rate_reset(self):
         self.assertEqual(p2.rate, 0.5)
         
         
+class TestSpitefulTitForTat(TestPlayer):
+    name = "Spiteful Tit For Tat"
+    player = axelrod.SpitefulTitForTat
+    expected_classifier = {
+        'memory_depth': 2,
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
 
+    def test_strategy(self):
+        """Starts by cooperating."""
+        self.first_play_test(C)
+
+    def test_effect_of_strategy(self):
+        """Repeats last action of opponent history until 2 consecutive defections, then always defects"""
+        self.markov_test([C, D, C, D])
+        self.responses_test([C] * 4, [C, C, C, C], [C])
+        self.responses_test([C] * 5, [C, C, C, C, D], [D])
+        self.responses_test([C] * 5, [C, C, D, D, C], [D])
```

Comment:
Could we add a test for the `self.retaliating` attribute directly and also a test that checks that when resetting the strategy this is reset as expected.

You can do my first request using the `responses_test`, which you can see here <http://axelrod.readthedocs.io/en/latest/tutorials/contributing/strategy/writing_test_for_the_new_strategy.html>:

> The function responses_test also accepts a dictionary parameter of attributes to check at the end of the checks. For example this test checks if the player’s internal variable opponent_class is set to "Cooperative":

So here you would have:

```
        self.responses_test([C] * 4, [C, C, C, C], [C], attrs={"retaliating": False})
        self.responses_test([C] * 5, [C, C, C, C, D], [D], attrs={"retaliating": False})
        self.responses_test([C] * 5, [C, C, D, D, C], [D], attrs={"retaliating": True}) 
```